docs/rules/forbid-elements.md

```diff
@@ -0,0 +1,39 @@
+# Forbid certain elements (forbid-elements)
+
+You may want to forbid usage of certain elements in favor of others, (e.g. forbid all `<div />` and use `<Box />` instead). This rule allows you to configure a list of forbidden elements and to specify their desired replacements.
+
+## Rule Details
+
+This rule checks all JSX components and verifies that no forbidden elements are used. When it detects a forbidden element, the error message will contain the specified elements that you should use instead. This rule is off by default. If on, no elements are forbidden by default.
+
+## Rule Options
+
+```js
+...
+"forbid-elements": [<enabled>, { "forbid": [<string|object>] }]
+...
+```
+
+### `forbid`
+
+An array of strings, objects, or a mixture of string and objects. If an item is a string, then an element matching the string will be forbidden and no replacement will be suggested. If an item is an object, then the keys will be the forbidden element and their values can be either a string or an array of strings of desired replacement elements.
+
+The following patterns are considered warnings:
+
+```jsx
+// [1, {forbid: ['button']}]
+<button />
+
+// [1, {forbid: ['button', 'input']}]
+<div><button /><input /></div>
+
+// [1, {forbid: [{button: 'Button']}]
```

Comment:
I like this suggestion. The only thing is I don't want to use the word `component` because to me that means a  react component which excludes simple DOM nodes. I like to use `element` because the it aligns with `React.createElement()`